File: rb/features.rb
# CelestrakGpData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module CelestrakGpDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      CelestrakGpDataBaseFeature.new
    when "ratelimit"
      CelestrakGpDataRatelimitFeature.new
    when "retry"
      CelestrakGpDataRetryFeature.new
    when "test"
      CelestrakGpDataTestFeature.new
    when "timeout"
      CelestrakGpDataTimeoutFeature.new
    else
      CelestrakGpDataBaseFeature.new
    end
  end
end

File: py/features.py
# CelestrakGpData SDK feature factory

from feature.base_feature import CelestrakGpDataBaseFeature
from feature.test_feature import CelestrakGpDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CelestrakGpDataBaseFeature(),
        "test": lambda: CelestrakGpDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
